_posts/2021-02-23-OpenSSL.markdown

```diff
@@ -0,0 +1,58 @@
+---
+title: Switching back to OpenSSL
+layout: post
+---
+
+The Void Linux team is planning on [performing the switch to
+OpenSSL](https://github.com/void-linux/void-packages/pull/21056) before March
+is over.
+
+For most users, this shouldn't be noticeable, since we will be pushing the same
+packages that already had to be rebuilt twice a year due to LibreSSL ABI
+breaks, but if anything in your setup depends explicitly on LibreSSL (you know
+who you are!), be sure to migrate away from that.
+
+The important parts of the post are done, but you can read on to learn about
+the decision process and some things we had to do.
+
+The discussion started in an
+[RFC](https://github.com/void-linux/void-packages/issues/20935) from April
+12th, 2020, and has been ongoing since then, with most maintainers expressing
+support for the idea. The main reasons for the switch are as listed in the RFC:
+
+- Less need for patching packaged software, which is made especially difficult
+  and potentially risky by the complexity of the OpenSSL API.
+- Better platform support for accelerated implementation on non `x86_64`
+  devices.
+- Access to new standards and algorithms earlier, such as full TLS 1.3 support.
+  Users of packages such as OpenVPN will benefit greatly from this.
+- More stable ABI, which saves builder time and user bandwidth.
+
+Since that time, Gentoo, which used to offer LibreSSL as an option, has
+[discontinued that
+support](https://www.gentoo.org/support/news-items/2021-01-05-libressl-support-discontinued.html),
+and LWN [covered the subject of LibreSSL on
+Linux](https://lwn.net/Articles/841664/) at the beginning of this year.
+
+Furthermore, Void's version of Qt5 is heavily patched in order to properly
+support LibreSSL, and work on porting Qt6 hasn't started yet, and is unlikely
+to be accepted by upstream in any manner. Other software, such as Python, is
+limited in the cyphers it can offer to running programs, since it depends on
+what the SSL library provides - they are even [considering dropping any support
+for LibreSSL in the
+future](https://www.python.org/dev/peps/pep-0644/#libressl-support).
+
+Unfortunately, this move also required us to drop some packages, which we had
+been able to carry until now due to LibreSSL providing an API between that of
+OpenSSL 1.0.1 and 1.1.1. Among these is software like Qt4.
+
+One great differentiating feature of LibreSSL is the `libtls` library it ships,
+which aims to be a simple interface for talking over the web through TLS, with
+sane defaults and hard to misuse. Some of the packages we ship depend on it, so
+there's work being done to package a [standalone version of
+it](https://github.com/void-linux/void-packages/pull/28732).
+
+I'd like to finish this post by saying that we greatly appreciate the work from
```

Comment:
The rest of this article is written from the perspective of "The Void Team" and you switch to "I' here.  Keep the voice the article starts with.  The closing is from the entire Void team, as we have greatly benefited from their work.  Maybe make a mention here that we continue to use other fine software from OpenBSD such as openssh and signify?

Reply:
The Void team is grateful for the work of the OpenBSD community on `[libressl-portable](https://github.com/libressl-portable/portable)`. Void continues to provide other excellent OpenBSD software, including OpenSSH and signify.

Reply:
Maybe also include links to OpenSSH and especially signify

Reply:
>  Void continues to provide other excellent OpenBSD software, including OpenSSH and signify.

signify is not only provided in the repo, but used for signing the iso images.  It think it should be mentioned.